backend/apps/core/tests/test_throttling.py

```
@@ -0,0 +1,238 @@
+            format='json'
+        )
+        # Should get 429 for rate limiting
+        if response.status_code == status.HTTP_429_TOO_MANY_REQUESTS:
+            self.assertIn('retry_after', response.json() or {})
+            self.assertIn('Retry-After', response.headers)
+
+    def test_login_rate_limit_headers(self):
+        """Test that login rate limit responses include proper headers."""
+        # Make requests until rate limited
+        responses = []
+        for i in range(12):
```

Ruff (0.14.3)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)